PayPal/src/main/java/com/braintreepayments/api/paypal/PayPalRequest.kt

```diff
@@ -0,0 +1,135 @@
+ *
+ * @see [Examples of prominent in-app disclosures](https://support.google.com/googleplay/android-developer/answer/9799150?hl=en.Prominent%20in-app%20disclosure)
+ *
+ * @property localeCode A locale code to use for the transaction.
+ * Supported locales are:
+ * <br></br>
+ * `da_DK`,
+ * `de_DE`,
+ * `en_AU`,
+ * `en_GB`,
+ * `en_US`,
+ * `es_ES`,
+ * `es_XC`,
+ * `fr_CA`,
+ * `fr_FR`,
+ * `fr_XC`,
+ * `id_ID`,
+ * `it_IT`,
+ * `ja_JP`,
+ * `ko_KR`,
+ * `nl_NL`,
+ * `no_NO`,
+ * `pl_PL`,
+ * `pt_BR`,
+ * `pt_PT`,
+ * `ru_RU`,
+ * `sv_SE`,
+ * `th_TH`,
+ * `tr_TR`,
+ * `zh_CN`,
+ * `zh_HK`,
+ * `zh_TW`,
+ * `zh_XC`.
+ *
+ * @property billingAgreementDescription Display a custom description to the user for a billing
+ * agreement. This property is optional.
+ * @property isShippingAddressRequired Whether to hide the shipping address in the flow. Defaults to
+ * false. When set to true, the shipping address selector will be displayed.
+ * @property isShippingAddressEditable Whether to allow the the shipping address to be editable.
+ * Defaults to false. Set to true to enable user editing of the shipping address. Only applies
+ * when [PayPalRequest.shippingAddressOverride] is set with a [PostalAddress].
+ * @property shippingAddressOverride a custom [PostalAddress], A valid shipping address to be
+ * displayed in the transaction flow. An error will occur if this address is not valid.
+ * @property landingPageType Use this option to specify the PayPal page to display when a user lands
+ * on the PayPal site to complete the payment.
+ * @property displayName The merchant name displayed in the PayPal flow; defaults to the company
+ * name on your Braintree account.
+ * @property merchantAccountId Specify a merchant account Id other than the default to use during
+ * tokenization.
+ * @property riskCorrelationId A risk correlation ID created with Set Transaction Context on your
+ * server.
+ * @property userAuthenticationEmail User email to initiate a quicker authentication flow in cases
+ * where the user has a PayPal Account with the same email.
+ * @property lineItems The line items for this transaction. It can include up to 249 line items.
+ */
+abstract class PayPalRequest internal constructor(
+    open val hasUserLocationConsent: Boolean,
+    open var localeCode: String? = null,
+    open var billingAgreementDescription: String? = null,
+    open var isShippingAddressRequired: Boolean = false,
+    open var isShippingAddressEditable: Boolean = false,
+    open var shippingAddressOverride: PostalAddress? = null,
+    open var landingPageType: PayPalLandingPageType? = null,
+    open var displayName: String? = null,
+    open var merchantAccountId: String? = null,
+    open var riskCorrelationId: String? = null,
+    open var userAuthenticationEmail: String? = null,
+    open var lineItems: List<PayPalLineItem> = emptyList()
+) : Parcelable {
+
+    enum class PayPalLandingPageType(val stringValue: String) {
```

Comment:
Do we want to declare a preference over enums living in their own classes or not? For the RBA metadata work, I started putting all the enums in their own files - but we can convert to this style if it's preferred

Reply:
Good callout. I was thinking the same. I'll move them out to their own classes/files.